mm2src/coins/lightning.rs

```diff
@@ -1593,14 +1686,81 @@ pub async fn get_claimable_balances(
     let ignored_channels = if req.include_open_channels_balances {
         Vec::new()
     } else {
-        ln_coin.channel_manager.list_channels()
+        ln_coin.list_channels().await
     };
-    let claimable_balances = ln_coin
-        .chain_monitor
-        .get_claimable_balances(&ignored_channels.iter().collect::<Vec<_>>()[..])
-        .into_iter()
-        .map(From::from)
-        .collect();
+    let claimable_balances = async_blocking(move || {
+        ln_coin
+            .chain_monitor
+            .get_claimable_balances(&ignored_channels.iter().collect::<Vec<_>>()[..])
+            .into_iter()
+            .map(From::from)
+            .collect()
+    })
+    .await;
 
     Ok(claimable_balances)
 }
+
+#[derive(Deserialize)]
+pub struct AddTrustedNodeReq {
+    pub coin: String,
+    pub node_id: PublicKeyForRPC,
+}
+
+pub async fn add_trusted_node(ctx: MmArc, req: AddTrustedNodeReq) -> TrustedNodeResult<String> {
+    let coin = lp_coinfind_or_err(&ctx, &req.coin).await?;
+    let ln_coin = match coin {
+        MmCoinEnum::LightningCoin(c) => c,
+        _ => return MmError::err(TrustedNodeError::UnsupportedCoin(coin.ticker().to_string())),
+    };
+
+    if ln_coin.trusted_nodes.lock().insert(req.node_id.into()) {
+        ln_coin.persister.save_trusted_nodes(ln_coin.trusted_nodes).await?;
+    }
+
+    Ok("success".into())
+}
+
+#[derive(Deserialize)]
+pub struct RemoveTrustedNodeReq {
+    pub coin: String,
+    pub node_id: PublicKeyForRPC,
+}
+
+pub async fn remove_trusted_node(ctx: MmArc, req: RemoveTrustedNodeReq) -> TrustedNodeResult<String> {
+    let coin = lp_coinfind_or_err(&ctx, &req.coin).await?;
+    let ln_coin = match coin {
+        MmCoinEnum::LightningCoin(c) => c,
+        _ => return MmError::err(TrustedNodeError::UnsupportedCoin(coin.ticker().to_string())),
+    };
+
+    if ln_coin.trusted_nodes.lock().remove(&req.node_id.into()) {
+        ln_coin.persister.save_trusted_nodes(ln_coin.trusted_nodes).await?;
+    }
+
+    Ok("success".into())
+}
+
+#[derive(Deserialize)]
+pub struct ListTrustedNodesReq {
+    pub coin: String,
+}
+
+#[derive(Serialize)]
+pub struct ListTrustedNodesResponse {
+    trusted_nodes: Vec<PublicKeyForRPC>,
+}
+
+pub async fn list_trusted_nodes(ctx: MmArc, req: ListTrustedNodesReq) -> TrustedNodeResult<ListTrustedNodesResponse> {
+    let coin = lp_coinfind_or_err(&ctx, &req.coin).await?;
+    let ln_coin = match coin {
+        MmCoinEnum::LightningCoin(c) => c,
+        _ => return MmError::err(TrustedNodeError::UnsupportedCoin(coin.ticker().to_string())),
+    };
```

Comment:
Since `lp_coinfind_or_err` always returns `MmCoinEnum` instance, we don't need to keep `coin` variable.

So we can shorten this like:

```rs
let ln_coin = match lp_coinfind_or_err(&ctx, &req.coin).await? {
    MmCoinEnum::LightningCoin(c) => c,
    e => return MmError::err(TrustedNodeError::UnsupportedCoin(e.ticker().to_string())),
};
```

Reply:
Done